snowflake/src/test/java/org/apache/iceberg/snowflake/JdbcSnowflakeClientTest.java

```diff
@@ -520,14 +757,12 @@ public void testGetTableMetadataSQLException() throws SQLException, InterruptedE
    */
   @Test
   public void testGetTableMetadataInterruptedException() throws SQLException, InterruptedException {
-    when(mockClientPool.run(any(ClientPool.Action.class)))
```

Comment:
after reading through the entire changes in this class I don't think we should be introducing `generateExceptionAndAssert` as this makes it more difficult to read the code